query.go

```diff
@@ -763,6 +764,22 @@ checkResponses:
 			// fixed before exposing this function for public use.
 			checkResponse(sm.sp, sm.msg, queryQuit)
 
+			// Each time we receive a response from the current
+			// peer, we'll reset the main peer timeout as they're
+			// being responsive.
+			if !peerTimeout.Stop() {
+				<-peerTimeout.C
+			}
+			peerTimeout.Reset(qo.timeout)
+
+			// Also at this point, if the peerConnectTimeout is
+			// still active, then we can disable it, as we're
+			// receiving responses from the current peer.
+			if connectionTicker != nil && !connectionTimeout.Stop() {
+				<-connectionTimeout.C
+			}
+			connectionTicker = nil
```

Comment:
Shouldn't we re-establish this ticker once we lose our current `queryPeer`?

Reply:
> As a result, we'll now _stop_ the now named `connectionTimeout` once we receive the first message from the peer. If the peer stops responding all together, then the `peerTimeout` will be ticked. Arguably, we don't even need the `connectionTimout` due to this new logic.

Why not just remove it then?

Reply:
Wanted to make the smallest change possible in this area to resolve the issue.